jupyterhub/app.py

```diff
@@ -855,7 +855,13 @@ def init_users(self):
         # but changes to the whitelist can occur in the database,
         # and persist across sessions.
         for user in db.query(orm.User):
-            yield gen.maybe_future(self.authenticator.add_user(user))
+            try:
+                yield gen.maybe_future(self.authenticator.add_user(user))
+            except Exception:
+                # TODO: known cause of this is system users that have been removed.
```

Comment:
Perhaps a bit more clarity in the comment. Not sure if I have the details correct.

# TODO: Review approach to synchronize whitelist with db
# known cause of the exception is trying to add a user who has already been removed from the system
# but the user still exists in the hub's user db